### PR TITLE
Closes #58

### LINKED ISSUE #58: Image/attachment upload
Still WIP in the Papercups backend, therefore blocked

### PR DESCRIPTION
**Pull request by ryg-git**

### CODE REVIEW

File: lib/widgets/sendMessage.dart [hunk 1]
```diff
@@ -77,6 +83,100 @@ class _SendMessageState extends State<SendMessage> {
     );
   }
 
+  Widget _getFilePicker() {
+    if (kIsWeb) {
```
Comment (aguilaair) [inline lib/widgets/sendMessage.dart:87]: Ideally we'd need an implementation that works on desktop too

File: lib/widgets/sendMessage.dart [hunk 2]
```diff
@@ -77,6 +83,100 @@ class _SendMessageState extends State<SendMessage> {
...
+                  widget.messages,
+                  widget.sending,
+                  attachments,
+                  fileIds,
+                  true,
+                );
+              }
+            }
+          } on PlatformException catch (e) {
+            print('sendMessage: Error in file picker: $e');
```
Comment (aguilaair) [inline lib/widgets/sendMessage.dart:140]: This is Ok for debugging, but let's make sure that the user gets feedback that something has gone wrong, possibly with a toast

File: lib/widgets/sendMessage.dart [hunk 3]
```diff
@@ -77,6 +83,100 @@ class _SendMessageState extends State<SendMessage> {
...
+              title: Text('Image'),
+              contentPadding: EdgeInsets.all(0),
+            ),
+          ),
+        ],
+      );
+    } else {
+      return IconButton(
+        icon: Icon(Icons.attachment_outlined),
+        onPressed: () {},
```
Comment (aguilaair) [inline lib/widgets/sendMessage.dart:175]: Let's let the user know that this feature is not currently supported on their platform, or directly hide it.

**Review by aguilaair — COMMENTED** [review]
Looking good. Just a few notes to improve the end user experience.

Thanks for your awesome work!

**Comment by ryg-git** [comment]
@aguilaair thanks for the review, I will implement the needful changes.

**Comment by aguilaair** [comment]
Ok, thank again for putting your time into this feature! I appreciate it.

**Review by aguilaair — APPROVED** [review]
Looks good, I'll do some minor changes and see of I can implement support for Windows and MacOS than i think it is ready to merge!

Thanks for you help 😃

**Review by aguilaair — CHANGES_REQUESTED** [review]
This implementation currently does not work with Web. `dart:io` is used and is therefore is incompatible with web. Additionally, the file picker will always return a `Uint8List` in web as paths aren't supported. If you have a chance to fix it it would be great!

**Comment by ryg-git** [comment]
I will fix it as soon as I can thanks for the  review @aguilaair

**Comment by aguilaair** [comment]
Thank you for your effort! I really appreciate it.

**Comment by ryg-git** [comment]
@aguilaair I have tested attachment upload and download on android device and on web (chrome), I am not able to test it on iOS device if it is possible for you to test it on iOS please do although I don't think it will not work on iOS. If you want anything changed please let me know.

Thank you!

**Comment by aguilaair** [comment]
Great! I'll check it out ASAP

**Review by aguilaair — APPROVED** [review]

**Comment by aguilaair** [comment]
Looks great! Made a few tewaks to show the body even if an attachment is shown and also tweaked some alerts to be more neutral themed.

Thanks for your effort. I believe this can now be merged and it will be included in the next release!

**Pull request merged**